Data on variety in bloggers (location, team)

catherinedevlin opened the pull request:
Pulls data from https://github.com/18f/data-private to report authors' locations and team, summarize on index page.

catherinedevlin committed:
Add bar charts for number of authors and author location

catherinedevlin committed:
Add missing pygal dependency

catherinedevlin committed:
Pull data on all authors from `private-data`
summarize authorship by location and team

Changed source of Author info to private-data repo

Calculate first_issue and issue-author links
Cleanup formatting with yapf and isort

Display info on blog author location, team

Format changes with yapf and isort

Remove unused import

fureigh commented:
@catherinedevlin I'll review this on Monday if nobody else jumps on it before then.

catherinedevlin committed:
Add Bokeh plot of issue lifecycles to report

Preliminary bokeh lifecycle plot is working

Terminate progress lines when issue closed

Reformat, remove unused imports

catherinedevlin committed:
Added bokeh to requirements

catherinedevlin committed:
Replace PyGal charts with Bokeh for consistency

catherinedevlin committed:
Initial commit

catherinedevlin committed:
Queries GitHub, serves Issue.lifecycle

catherinedevlin committed:
Serves chart with tap and hover

catherinedevlin committed:
Accept arbitrary owner/repo

catherinedevlin committed:
Add API

catherinedevlin committed:
trying to make requests.get responsive to whether issues or events are requested

catherinedevlin committed:
basic passing tests

catherinedevlin committed:
Test for persistence added

catherinedevlin committed:
New tests for using cached data or refreshing

catherinedevlin committed:
Share code between API and Bokeh functions

Share code between API and chart output

Add beginning README

catherinedevlin committed:
flask-restful is required

catherinedevlin committed:
Removed issue lifecycle charts

catherinedevlin committed:
Embedding chart from included github_issue_lifecycles

catherinedevlin committed:
Add 'app/github_issue_lifecycles/' from commit '49ae1d98d3da37acd76cce0a3f10c3dd58e06cfc'

git-subtree-dir: app/github_issue_lifecycles
git-subtree-mainline: 24037c4984d391c993a2906fdadc0e396520332f
git-subtree-split: 49ae1d98d3da37acd76cce0a3f10c3dd58e06cfc

catherinedevlin committed:
format and pdb cleanup

catherinedevlin committed:
Automatically assign colors when fetching repo

catherinedevlin committed:
git subtree used to include github_issue_lifecycle project

catherinedevlin committed:
Python to 3.4

catherinedevlin committed:
Python 3.5

catherinedevlin committed:
Incorporating cleanup and coverage updates from github_issue_lifecycle

Squashed commit of the following:

commit c31a753b904799c57a7468bf590a280e8be3bb6f
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Tue Dec 29 11:41:40 2015 -0500

    Increase test coverage

commit 6181e8d9756f3f51eb04a51eb09e0c0ccdeadd54
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 22:58:07 2015 -0500

    Install codecov for travis

commit 9b3713d651ba29b2de8c1f5744dce6c1e432e420
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 18:51:55 2015 -0500

    set testing environment in testing

commit b0c54c779fbd916dc91773dec83188ff19289475
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 18:21:58 2015 -0500

    create test database for travis

commit 031e0ec035c578fee13cc472788c98a248c32ba4
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 17:57:16 2015 -0500

    Include requirements.txt explicitly

commit 1752d77ae52baf6a93d1e72dd240c53fea69d3c9
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 16:48:05 2015 -0500

    dev-requirements.txt added

commit 28bf5d9fa08f6b7c5bd7d175a3f441f54476aa89
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 16:40:51 2015 -0500

    travis apparently demands a setup.py

commit ea8eca72a85d376996368ea10a09035f51c6485f
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 14:09:13 2015 -0500

    travis fix: No setup.py

commit aa86eec14dec75e34b33b4168d62e7a13fd91df6
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 14:00:21 2015 -0500

    Strip unwanted parts of .travis.yml

commit d7606aab5dc0ead6811540c4385275c10b8b1431
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Mon Dec 28 13:41:12 2015 -0500

    Cleanup of formatting and unused code

commit da33f121930b3ca8aec4a8c032037e228f8f0e5e
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Thu Dec 24 09:56:12 2015 -0500

    Update color map when fetching repo

commit 3569ccddcd52c7ff3f2a74c1a463b2d50c7de115
Author: Catherine Devlin <catherine.devlin@gsa.gov>
Date:   Tue Dec 22 12:05:47 2015 -0500

    Add default port, strip unused authentication code, cycle colors in case of many milestones

    Add default port.  Strip unused code.

    Strip unused authentication code

    Cycle colors to keep from running out

catherinedevlin commented:
I've done a bunch more work on this in the course of pulling the graph of github issues out into a freestanding repo (https://github.com/18F/github-issue-lifecycle, incorporated here with a git subtree).  Should be ready for review and merge!

Commit:
Merge branch 'master' into author_variety

Commit:
3.5 version
